json-core/src/main/java/io/micronaut/json/convert/JsonConverterRegistrar.java

```diff
@@ -114,13 +123,19 @@ public void register(ConversionService<?> conversionService) {
     public TypeConverter<JsonArray, Iterable> arrayNodeToIterableConverter() {
         return (node, targetType, context) -> {
             Map<String, Argument<?>> typeVariables = context.getTypeVariables();
```

Comment:
minor: `typeVariables` may be moved down so that it's not allocated in the 3rd/default case.